cmd/sink/main.go

```diff
@@ -94,6 +98,46 @@ func (sink *Sink) Receive(ctx context.Context, event cloudevents.Event) {
 		event.ID(),
 		k8sObj.GetUID(),
 	)
+	isDeleteType, err := sink.IsDeleteExpr.Evaluate(event)
+	if err != nil {
+		sink.Logger.Printf("Could not determine if the type cloudevent %s is 'delete': %s\n", event.ID(), err)
+		return
+	}
+
+	if ceSql.BooleanType.IsSameType(isDeleteType) {
+		sink.Logger.Printf(
+			"The type of cloudevent %s is Delete. Checking if object %s needs to be archived\n",
+			event.ID(),
+			k8sObj.GetUID(),
+		)
+		boolVal, isBool := isDeleteType.(bool)
+		if !isBool {
+			sink.Logger.Printf(
+				"Could not convert result of the cesql expression to boolean: %s\n",
+				err,
+			)
+			return
+		}
+		if boolVal && sink.Filters.MustArchiveOnDelete(ctx, k8sObj) {
+			ctx, cancel := context.WithTimeout(ctx, time.Second*5)
+			err = sink.Db.WriteResource(ctx, k8sObj, event.Data())
+			if err != nil {
+				sink.Logger.Printf(
+					"Failed to write object %s from cloudevent %s to the database: %s\n",
+					k8sObj.GetUID(),
+					event.ID(),
+					err,
+				)
+			}
+			defer cancel()
+			return
+		}
+		sink.Logger.Printf(
+			"Object %s from cloudevent %s does not need to be archived after deletion\n",
+			k8sObj.GetUID(),
+			event.ID(),
+		)
```

Comment:
It looks like delete events will still go through the code for updates and creates.

Reply:
because i forgot the return at the end of the if. I didn't have time to test the changes yesterday which is why I marked this as a draft :smiley: